### ISSUE TITLE
Unable to destroy resources in workspace after workspace removal warning

### ISSUE THREAD

**Issue by pselle** [hashicorp/terraform#24019]
### Terraform Version

```
0.12.17
```

Built from https://github.com/hashicorp/terraform/pull/22540

### Terraform Configuration Files

```hcl


terraform {
  backend "cos" {
    region = "na-ashburn"
    bucket = "bucket-for-terraform-state-xxxxxx"
    prefix = "terraform/state"
  }
}

resource "null_resource" "name" {
  for_each = toset(local.x)
}

locals {
  x = ["a","b","c"]
}


```

### Debug Output
https://gist.github.com/pselle/6f8dc4f1ae686398f92b2b6a9a7aade9

### Expected Behavior

Should be able to destroy objects in workspace

### Actual Behavior

State lock is lingering

### Steps to Reproduce

1. Create a new workspace (ex. `space3`) populated with objects
2. `terraform workspace select default` go to a different workspace
3. `terraform workspaces delete space3` try to delete the workspace, you'll get an error that there are objects remaining, and that it could create lingering objects.
```
$ terraform workspace delete space3
Acquiring state lock. This may take a few moments...
Workspace "space3" is not empty.

Deleting "space3" can result in dangling resources: resources that
exist but are no longer manageable by Terraform. Please destroy
these resources first.  If you want to delete this workspace
anyway and risk dangling resources, use the '-force' flag.
```
4. `terraform workspace select space3`
5. `terraform destroy` Unable to destroy objects, as a state lock is present. This is present through multiple runs/is consistent.

### Additional Context

The workflow above is should be able to re-create the error consistently. In addition to being unable to destroy objects in `space3`, any interaction in the workspace `space3` is now blocked by the lingering state lock (cannot `apply` either)

### References

- https://github.com/hashicorp/terraform/pull/22540

**Comment by pselle** [comment]
Additionally, after running `destroy lock=false`, in order to run the destroy anyway, and then deleting the workspace `terraform workspace delete space3`, I cannot create a new workspace with the same name:

```
$ terraform workspace new space3
Failed to lock cos state: failed to create tag: tencentcloud-terraform-lock -> 2846f8499a5d2a0809dcf783cceea4d6: [TencentCloudSDKError] Code=ResourceInUse.TagDuplicate, Message=[SWOOLE_WTAG] tagKey-tagValue have exists., RequestId=f0c4602b-066c-47df-8678-e14b09aa5f0b
Lock Info:
  ID:        9dda333eb4a807d9f1b1efc0126c42e1
  Path:      terraform/state/space3/terraform.tfstate.tflock
  Operation: workspace_delete
  Who:       pamselle@Pam-MBP.local
  Version:   0.12.17
  Created:   2020-02-03 16:45:48.151353 +0000 UTC
  Info:      

$ terraform workspace list
  default
* space2

```

**Comment by likexian** [comment]
Hello @pselle 
Thanks for your testing and issue, I have debug this issue carefully and finally found that this is not the issue of `tencentcloud backend` but the bug of `terraform core`, the locker is not released after workspace removal warning.
I have fired a PR, but I am not sure whether my PR is the best solution, please have a review, thanks.

**Comment by likexian** [comment]
This has been resolved, please close it.

**Comment by pselle** [comment]
This will be included in the next 0.12 release.
